### DISCUSSION TITLE
Is Huawei Luna 2000 supported?

### DISCUSSION THREAD

**Discussion by darmach** [wlcrs/huawei_solar#793] [Setting up the integration]
Hi! I've been offered HUAWEI SUN 2000-6KTL-M1  inverter with Huawei Luna 2000-7KWh-C1 storage, Huawei DTSU-666-H counter and Luna 2000 batteries. Is it supporter by this integration? Can it be scraped locally via modbus?

**Comment by tmenguy** [comment]
I have a LUNA (and 3 modules of those) works like a charm with the integration

**Reply by Roving-Ronin** [reply]
Re the inverter see: https://github.com/wlcrs/huawei_solar/wiki/Confirmed-firmware-versions#three-phase---sun2000-m1-hybrid
In regards the LUNA I note this is the new LUNA S1 (7/14/21kWh) model ( jealous ;-) ), that I haven't see much comment upon people saying they have installed. However, it should work fine given it using the same modbus registers.

In regards the DTSU-666-H smart/consumption meter, if you ever intend to charge an EV I suggest you see if you can get them install teh Huawei sCharger now and to instead install a DTSU-666-FE (Fast Ethernet). 
This is a newer model that adds Fast Ethernet connectivity, allowing it to communicate to the Inverter AND a Huawei (EV) sCharger. If you don't have this newer meter and install the sCharger, it will either require the installation of the DSTU-666-FE (in parallel to the existing DSTU-666-H) or otherwise the sCharger will end up connecting to the inverter and using the sDongle to provide it with a 'virtual meter', that uses the single modbus-tcp connection the sdongle provides. If this happens, you will NOT be able to run the sCharger and connect the inverter/battery to Home Assistant.

See: https://wsolar.shop/en/huawei-smart-power-sensor-dtsu666-fe-3ph-1ph-100a

Huawei connectivity scenarios: https://support.huawei.com/enterprise/en/doc/EDOC1100280349/3e606caf/solution-scenarios

WRT to the inverter, the M1 this is a current model but there is also a newer range the MB0 that is now available. They are larger capacity inverters, but also they support load balancing across all 3 phases (which M1 don't) and also support (easier) running of a backup generator to backup the house in an outage. The balancing of phases is handy to have, especially if looking at adding SmartGuard at some point, that will provide backup battery operation with the LUNA if there is a grid outage.

https://www.vpsolar.com/en/new-huawei-mb0-series-three-phase-hybrid-inverters-and-storage-system/

**Reply by tmenguy** [reply]
Hi,

Thx on my side I do have the MB0 , and The new LUNA S1 (21kWh) ... but as of the SmartGuard and from another thread ... it is only backing up on 1 phase no?.

As for the sCharger I have a Wallbox and did a software to adapt the charge current according to solar/battery power from this great HA integration ..... but instead of this DSTU-666-FE, don't you think an EMA may make more sense?

BR
Thomas

**Reply by Roving-Ronin** [reply]
Is that the more recently announced SmartGuard for 3 phase, not the 'original' one?
Can't remember the overseas suffix but in Australia it's SmartGuard ending in 'AUT0'.
Works with M1 but only protects 1 circuit, whereas with MB0 that can balance between phases, it's supposed to allow (looking at wiring diagrams) protecting one phase during the grid drop and then using an auto disconnect/transfer switch to drop off grid and then work all 3 phases of battery or a generator (including auto start up).

What price are the 7/14/21 LUNA selling for in (which country) and what's the price for the old 5/10/15kW LUNA ?

**Comment by tmenguy** [comment]
Hum interesting from the diagrammes it seems it is only the MAP0 that can do the 3 phases, and the M1/MB0 cannot … but if you have a diagram showing something different, I would be super happy! Can you point me to the right one? I probably missed it.

The price in France are not public but super expensive: I paid 10k euros without taxes so 12k for 3 modules of S1 (so 21kWh).

 BR
Thomas

**Reply by Roving-Ronin** [reply]
I think your right and I'm confusing my MB0 vs the MAP0 🤔🤦‍♂️

€12k / AUD$19.5k ouch... we are still only seeing the S0 here and ~€9.5k / AUD$15k for the 15kWh.

Begs the question of why prices remain (artificially) high whilst the cost of lithium has halved in the last year.
